Magento2/Tests/PHP/AutogeneratedClassNotInConstructorUnitTest.1.php.inc

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types = 1);
+
+namespace Magento2\Tests\PHP;
+
+use Magento2\Model;
+
+class Good
+{
+    public function __construct(
+        Model $model = null
+    )
+    {
+        $this->model = $model ?? ObjectManager::getInstance()->get(Model::class);
+    }
+
+    /**
+     * @return Model
+     */
+    public function otherMethodThatCallsGetInstanceBad(): void
+    {
+        $model = ObjectManager::getInstance()->get(Model::class);
+        $model->something();
+    }
+
+    /**
+     * @return Model
+     */
+    public function otherMethodThatCallsGetInstanceGood(): void
+    {
+        $model = $this->model ?? ObjectManager::getInstance()->get(Model::class);
+        $model->something();
+    }
```

Comment:
There is another case missing here, what about assigning an object manager instance to a variable?

```suggestion
    }
    
    /**
     * @return Model
     */
    public function objectManagerAssignedToAVariableBad(): void
    {
        $objectManager = ObjectManager::getInstance();
        $model = $objectManager->get(Model::class);
        $model->something();
    }
```

Reply:
This is not on the ticket.